### PR TITLE
Add test development for JWT token x5c

### PR DESCRIPTION
**Pull request by kshpak**
In the future, there will be a complete security scenario with Quarkus + Vert.x and JWT tokens
TD for https://issues.redhat.com/browse/QUARKUS-579

### CODE REVIEW

File: 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                "fmxP1Ow3QPZJfHaHj+aXddWZGOD5cP5karziVNZUYmEPqNrrs5MlSCPYGdFSp+wN\n" +
+                "Vt8+8lZ2/YGppkV/88t1zYYQ6UxisQt4q/u25wJ+j9fFDDnYq7Mt/orCMrKODBe4\n" +
+                "TNFkafBY6Jo/";
+
+        new JWT().addJWK(new JWK(new JsonObject()
+                .put("kty", "RSA")
+                .put("alg", "RS256")
+                .put("x5c", new JsonArray()
+                        .add(certificateOne)
+                .add(certificateTwo))));
```
Comment (Sgitario) [inline 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java:61]: Can you fix the indentation here to improve the readability? 
```
new JWT().addJWK(new JWK(new JsonObject()
                .put("kty", "RSA")
                .put("alg", "RS256")
                .put("x5c", new JsonArray()
                        .add(certificateOne)
                        .add(certificateTwo))));
```

File: 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                "goykwIMlR7jEg7bIwoidrvG+SzAfBgNVHSMEGDAWgBT/goykwIMlR7jEg7bIwoid\n" +
+                "rvG+SzAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQCJJj0Pa6Xi\n" +
+                "M5B3PMZjx/3Bb5SVUqiklGTYxpql4w4l7v82EBFDB+NdtkILNbzjxZyfbdz17Bv8\n" +
+                "uo4/BUCIgvmHKBuYR54Ie1rjEe2dSMIm0pYQBetOLNU0ukV1yJ+aawzv680KBVvz\n" +
+                "eDs2m8EJ0BUysh+1JTWWI5nJ1mbjVaSVDylnNRvaN9cUGTBKvGQtXMTCracXHIg4\n" +
+                "fmGug3T9GsswJorX+xn3O0jPpoYAouHfMhKkyx2lXdm/7oq2HlXc6QOylxZ4FUEr\n" +
+                "pUJnaHA2ErjRa5kHvP5aqEvmSnKGfDB+8XvIUUDnl3R6QDGkzdTjlFHC4lQAITxq\n" +
+                "7huYsdhAMap2";
+
+        String certificateTwo = "MIIDZTCCAk2gAwIBAgIUBD3+ujEHrvjX67xTgBUzSecAiRMwDQYJKoZIhvcNAQEL\n" +
```
Comment (Sgitario) [inline 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java:36]: I think that it makes sense to copy out the certificates in separate files as the content is not relevant for the test.
This way the code would look like as:
```java
new JWT().addJWK(new JWK(new JsonObject()
                .put("kty", "RSA")
                .put("alg", "RS256")
                .put("x5c", new JsonArray()
                        .add(fromResource("one.cert"))
                        .add(fromResource("two.cert")))));
```

**Review by Sgitario — CHANGES_REQUESTED** [review]
The test is not working in Native mode which is kind of expected as this is not a Quarkus application.

File: 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java (hunk 1 above)
Comment (kshpak) [reply 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java]: Fixed

File: 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java (hunk 2 above)
Comment (kshpak) [reply 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java:36]: For me, it's easier to read with String variables, I would leave it as is. Anyway later it should be converted from resource file to string

Comment (rsvoboda) [reply 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java:36]: @kshpak please add `//TODO convert from string to resource file`

**Review by rsvoboda — CHANGES_REQUESTED** [review]
add following snippet into 302-quarkus-vertx-jwt/pom.xml
```xml
  <profiles>
    <profile>
      <id>native</id>
      <activation>
        <property>
          <name>native</name>
        </property>
      </activation>
      <properties>
        <quarkus.package.type>none</quarkus.package.type>
      </properties>
    </profile>
  </profiles>
```

This will ensure "skip" in native mode.

File: 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java (hunk 2 above)
Comment (kshpak) [reply 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/JwtTest.java]: Resolved

**Comment by kshpak** [comment]
> add following snippet into 302-quarkus-vertx-jwt/pom.xml
> 
> ```
>   <profiles>
>     <profile>
>       <id>native</id>
>       <activation>
>         <property>
>           <name>native</name>
>         </property>
>       </activation>
>       <properties>
>         <quarkus.package.type>none</quarkus.package.type>
>       </properties>
>     </profile>
>   </profiles>
> ```
> 
> This will ensure "skip" in native mode.

Resolved

**Review by Sgitario — APPROVED** [review]

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**